### ISSUE TITLE
可以在scim的输入法插件中使用libdbus的api监听其他应用程序发送的dbus信号吗?

### ISSUE THREAD

**Issue by hantengc** [scim-im/scim#43]
scim本身是基于socket实现的事件循环，我想要在scim的输入法引擎插件中实现一套libdbus的api用来监听信号
`        std::thread t([this]()
                      {
                while (1)
    {
        dbus_connection_read_write_dispatch(dbus_connection, -1);
    } });

        t.detach();`

也有单开线程防止阻塞主事件循环，但是还是收不到信号。

请问scim对其是有什么限制吗？是否有其他方法可以做到呢？

**Comment by hantengc** [comment]
@tzhuan

**Comment by tzhuan** [comment]
The SCIM project was lunched in 2002, earlier than the D-Bus project (2006), so I don't think SCIM has any reason or intent to limit the use of D-Bus. Since I completely don't understand D-Bus, I can't provide meaningful advice.

**Comment by hantengc** [comment]
> The SCIM project was lunched in 2002, earlier than the D-Bus project (2006), so I don't think SCIM has any reason or intent to limit the use of D-Bus. Since I completely don't understand D-Bus, I can't provide meaningful advice.

scim的事件循环是依赖的socket吗？

**Comment by tzhuan** [comment]
The scim socket event loop is inside the `SocketServer::run()`: https://github.com/scim-im/scim/blob/1df63e9f0317c5dc5a834f8f2ec3f3da148551f1/src/scim_socket.cpp#L849
